sfge/src/main/java/com/salesforce/rules/unusedmethod/StaticMethodCallValidator.java

```diff
@@ -36,11 +41,142 @@ protected boolean internalUsageDetected() {
      *   <li>Any class that extends the class that defines the method.
      * </ol>
      *
-     * @return
+     * @return True if external usage found, else false
      */
     @Override
     protected boolean externalUsageDetected() {
-        // TODO: IMPLEMENT THIS METHOD.
+        return usageDetected(new ExternalUsageDetector());
+    }
+
+    private boolean usageDetected(UsageDetector detector) {
+        // First, use the detector to check for usage via the host class.
+        if (detector.detectsUsage(targetMethod.getDefiningType())) {
+            return true;
+        }
+
+        // If the method is heritable, we also need to check for usage in subclasses.
+        if (methodIsHeritable()) {
+            // We'll start with a list of just the immediate subclasses of the host class.
+            List<String> subclassNames =
+                    ruleStateTracker.getSubclasses(targetMethod.getDefiningType());
+            // Using a while-loop instead of a for-each loop lets us grow the list as we iterate
+            // through it.
+            int i = 0;
+            while (i < subclassNames.size()) {
+                String subclassName = subclassNames.get(i);
+                // If we find a usage in that subclass, we're done.
+                if (detector.detectsUsage(subclassName)) {
+                    return true;
+                }
+                // Otherwise, if the subclass is itself extensible, then its own subclasses must be
+                // added for consideration.
+                subclassNames.addAll(ruleStateTracker.getSubclasses(subclassName));
```

Comment:
Sounds like Stack would be a good fit to push more values and pop as they get handled.

Reply:
Using a `Stack` would make the search depth-first, so I chose a `List` to keep the search breadth-first.